src/sqlitestorage.h

```diff
@@ -531,6 +531,8 @@ public Q_SLOTS:
   }                                                     \
 }
 
+#define CREATE_METADATA \
+  "CREATE TABLE IF NOT EXISTS Metadata(transactionId INTEGER)"
```

Comment:
Metadata is a generic concept but I suppose this wouldn't be easy to extend to include anything else than the transactionId. Could then just call this with transaction naming?

A table with always one row feels interesting way to store one integer but don't have immediately better alternatives to offer. Guess it does the trick. And if there's a better way, could be easy to just ditch this without data migration

Reply:
> Metadata is a generic concept but I suppose this wouldn't be easy to extend to include anything else 

Well, the idea was to use `ALTER TABLE Metadata ADD COLUMN` or friends to add more metadata as needed.

> A table with always one row feels interesting way to store one integer but don't have immediately better alternatives to offer.

Indeed, I crawl over forums here and there to see how this idea can be achieved, or if SQlite provides some way of storing global variables, but didn't find anything else…

Reply:
> Well, the idea was to use ALTER TABLE Metadata ADD COLUMN or friends to add more metadata as needed.

Was myself thinking the update difficulty from all the rows needing then this transaction id. But indeed it could work if there's never going to be more rows here. Just a single one with more columns for different things.

Reply:
Yes, the idea of this metadata table is to store global values for the database, values being defined by column, with only ever one single row. Then, the `selectMetadata()` and `updateMetadata()` API could take more arguments to get or set the additional value in this single row.

I've no idea if SQlite is providing a better way to do this.